File: python/minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        minimal, count = k, 0
        for i in range(len(blocks)):
            if k <= i:
                minimal = min(minimal, count)
            if blocks[i] == "W":
                count += 1
            if k <= i and blocks[i - k] == "W":
                count -= 1
        return min(minimal, count)

File: rust/minimum-recolors-to-get-k-consecutive-black-blocks.rs
use std::cmp;

impl Solution {
    pub fn minimum_recolors(blocks: String, k: i32) -> i32 {
        let mut count:i32 = 0;
        let mut minimal:i32 = k;
        for i in 0..blocks.len() {
            if k as usize <= i {
                minimal = cmp::min(count, minimal);
            }
            let c:char = blocks.chars().nth(i).unwrap();
            if c == 'W' {
                count += 1;
            }
            if k as usize <= i  {
                let d:char = blocks.chars().nth(i - (k as usize)).unwrap();
                if d == 'W' {
                    count -= 1;
                }
            }
        }
        cmp::min(count, minimal)
    }
}
